### ISSUE TITLE
Border radius control: incorrect usage of Tooltip causes unexpected tab stop

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#68591]
### Description

Discovered while testing https://github.com/WordPress/gutenberg/pull/68498

This is one more occurrence of incorrect usage of the Tooltip component. Cc @WordPress/gutenberg-components 

The Ariakit-based Tooltip adds a `tabindex=0` attribute to the DOM element it wraps. When the wrapped DOM element happens to be an element that is supposed to be focusable that't not a big deal. However, when the wrapped element is just a DIV element, it becomes a focusable non-interactive, non-labeled element. This isn't OK and among other things it causes an unexpected tab stop when using the keuboard.

https://github.com/WordPress/gutenberg/blob/ddfbdcc33538d6522d64e517ecd02f7952c28d6f/packages/block-editor/src/components/border-radius-control/input-controls.js#L71-L86

In this case, the element with CSS class `components-border-radius-control__tooltip-wrapper` gets a `tabindex=0`. As such, tabbing through the UI requires an additional, unexpected, Tab press on an element that isn't interactive just to make hte Tooltip appear.

Animated GIF:

![Image](https://github.com/user-attachments/assets/95ad3886-d9c7-4d9b-a1bc-8cb43ca68014)

### Step-by-step reproduction instructions

- Select a Group block.
- Go to the block settings panel > Styles > Radius setting and click the 'Unlink radii' icon button to switch the control to the UI that renders 4 input fields.
- Use the Tab key to navigate through the four input fields.
- Observe you have to Tab three times for each input:
  1. FIrst tab stop: the focusable DIV with tabindex=0
  2. Second tab stop: the actual input field
  3. Third tab stop: The 'Select unit' select element.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by Rishit30G** [comment]
Thanks for sharing the issue, 
I could replicate the issue on my local 
So as per my understanding if we remove the div and add the class to the UnitControl Component, we can get proper tab navigation 

```jsx
<Tooltip text={ label } placement="top" key={ corner }>
     <UnitControl
		{ ...props }
		aria-label={ label }
		value={ [ parsedQuantity, computedUnit ].join('') }
		onChange={ createHandleOnChange( corner ) }
		onUnitChange={ createHandleOnUnitChange( corner ) }
	         size="__unstable-large"
		className="components-border-radius-control__tooltip-wrapper"
	/>
</Tooltip>
```

Before: 

https://github.com/user-attachments/assets/76bce312-2d94-41ec-a136-583398b3630a


After: 

https://github.com/user-attachments/assets/9f6245c2-f188-416d-9118-d4af8c0354b5

**Comment by yogeshbhutkar** [comment]
I think we can pass the classname to `Tooltip` component directly as `Unit Control` is indeed what is getting wrapped here.

Edit: Looks like the `classname` isn't used anywhere.

**Comment by afercia** [comment]
I think this should be fixed by removing the Tooltip component, set the `key` to the wrapepr and change `aria-label` to `label` on the `UnitControl`. 

This would make the input field labels visible. As such, some design adjustments may be neeed but input fields are really not supposed to have tooltips. 

My guess is that the current, incorrect, implementation was only done to hide the labels and use tooltips for the input fields. That's not great for accessibility.

**Comment by yogeshbhutkar** [comment]
I believe using `help` instead of both `aria-label` and `label` could be better from the UI perspective.

Difference:

|`label`|`help`|
|-|-|
|![Image](https://github.com/user-attachments/assets/7323ecd5-eafa-4a60-a40f-badae28e54c8)|![Image](https://github.com/user-attachments/assets/7ac52bc3-1c6b-4dc2-9c36-68a3ed2cb0f8)|
